pkgs/development/libraries/qt-6/modules/qmake2cmake.nix

```diff
@@ -0,0 +1,68 @@
+/*
+sample use
+nixpkgs/pkgs/tools/misc/qarma-for-qt6/default.nix
+*/
+
+{ python3
+, python3Packages
+, git
+, srcs
+}:
+
+python3Packages.buildPythonPackage rec {
+  pname = "qmake2cmake";
+  inherit (srcs.qtbase) src version;
+  sourceRoot = "qtbase-everywhere-src-${version}/util/cmake";
+
+  propagatedBuildInputs =
+    (with python3Packages; [ pyparsing portalocker sympy pytest ])
+    ++ [ git ];
+
+  patchPhase = ''
+    mv pro2cmake.py ${pname}
+    ln -s ${pname} pro2cmake.py # for tests
+
+    cat >setup.py <<'EOF'
+    #! ${python3.interpreter}
+    from distutils.core import setup
+    setup(
+      name='${pname}',
+      version='${version}',
+      #packages=['distutils', 'distutils.command'],
+      scripts=['${pname}'],
+    )
+    EOF
+
+    # fix: read-only filesystem
+    sed -i 's|cache_path = get_cache_location()|cache_path = "/tmp/${pname}-cache.json"|' condition_simplifier_cache.py
+
+    # copy dependencies
+    mkdir -p $out/bin
+    for f in *.py
+    do
+      cp $f $out/bin/
+    done
+    chmod -x $out/bin/*.py
+
+    # fix tests
+    export HOME=/tmp
+  '';
+
+  # use original name as alias
+  postInstall = ''
+    rm $out/bin/pro2cmake.py
+    ln -s ${pname} $out/bin/pro2cmake.py
```

Comment:
This should be cp or we reference the sources at runtime and cannot garbage collect them.

Reply:
> This should be cp

wdym?

Reply:
Instead of symlinking (ln) we should copy (cp) or move (mv) the file so that the final system has no reference to the src anymore.

Reply:
> so that the final system has no reference to the src anymore.

there is no reference to src


<pre>
$ ls -l result/bin/
total 188
-r--r--r-- 1 root root  4886  1. Jan 1970  cmakeconversionrate.py
-r--r--r-- 1 root root  6513  1. Jan 1970  condition_simplifier_cache.py
-r--r--r-- 1 root root  9278  1. Jan 1970  condition_simplifier.py
-r--r--r-- 1 root root 54946  1. Jan 1970  configurejson2cmake.py
-r--r--r-- 1 root root 37268  1. Jan 1970  helper.py
-r--r--r-- 1 root root  3991  1. Jan 1970  json_parser.py
lrwxrwxrwx 1 root root    11  1. Jan 1970  <span color="#55FFFF"><b>pro2cmake.py</b></span> -&gt; <span color="#55FF55"><b>qmake2cmake</b></span>
-r--r--r-- 1 root root  8156  1. Jan 1970  pro_conversion_rate.py
-r-xr-xr-x 1 root root   516  1. Jan 1970  <span color="#55FF55"><b>qmake2cmake</b></span>
-r--r--r-- 1 root root 14832  1. Jan 1970  qmake_parser.py
-r--r--r-- 1 root root  8286  1. Jan 1970  run_pro2cmake.py
-r--r--r-- 1 root root   233  1. Jan 1970  setup.py
-r--r--r-- 1 root root 16254  1. Jan 1970  special_case_helper.py
</pre>

`pro2cmake.py` is just an alias for `qmake2cmake`
since `pro2cmake.py` is the original name of the script

```
$ ./result/bin/pro2cmake.py 2>&1 | head -n 1
usage: qmake2cmake [-h] [--debug] [--debug-parser] [--debug-parse-result]

$ ./result/bin/qmake2cmake 2>&1 | head -n 1
usage: qmake2cmake [-h] [--debug] [--debug-parser] [--debug-parse-result]
```